src/k2/module/k23si/Module.cpp

```diff
@@ -272,6 +277,227 @@ bool K23SIPartitionModule::_validateStaleWrite(dto::K23SIWriteRequest& request,
     return true;
 }
 
+std::size_t K23SIPartitionModule::_findField(const dto::Schema schema, k2::String fieldName ,dto::FieldType fieldtype) {
```

Comment:
Not related with new code, but 10 lines above, we have
"else if (versions.size() > 1 && versions[0].status == dto::DataRecord::WriteIntent &&
        request.mtr.timestamp.compareCertain(versions[1].txnId.mtr.timestamp) <= 0) {"
This optimization is good/correct, but if re-write of same KV in the same transaction is not over-write the same dataRecord in the Deque, this line will cause the re-write fail on the third time in a transaction. A bug scenario Yazhi mentioned in previous meeting.